OPAL/apk/src/main/scala/org/opalj/apk/parser/ApkParser.scala

```diff
@@ -0,0 +1,354 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package apk
+package parser
+
+import com.typesafe.config.Config
+import net.dongliu.apk.parser.ApkFile
+import org.opalj.apk.ApkComponent
+import org.opalj.apk.ApkComponentType
+import org.opalj.apk.ApkComponentType.ApkComponentType
+import org.opalj.apk.ApkComponentsKey
+import org.opalj.apk.parser.DexParser.DexParser
+import org.opalj.br.analyses.Project
+import org.opalj.ll.LLVMProjectKey
+import org.opalj.log.GlobalLogContext
+import org.opalj.log.LogContext
+import org.opalj.log.OPALLogger
+import org.opalj.util.PerformanceEvaluation.time
+
+import java.io.File
+import java.io.StringWriter
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters.EnumerationHasAsScala
+import scala.sys.process.ProcessLogger
+import scala.sys.process.stringToProcess
+import scala.xml.Node
+import scala.xml.XML
+/**
+ * Parses an APK file and generates a [[Project]] for it.
+ *
+ * The generated [[Project]] contains the APK's Java bytecode, its native code and its components / entry points.
+ *
+ * Following external tools are utilized:
+ *   - enjarify or dex2jar (for creating .jar from .dex)
+ *   - RetDec (for lifting native code to LLVM IR)
+ *
+ * @param apkPath path to the APK file.
+ *
+ * @author Nicolas Gross
+ */
+class ApkParser(val apkPath: String)(implicit config: Config) {
```

Comment:
Is this always created via the `createProject` method? In that case, the constructor should maybe be private